Dockerfile

```diff
@@ -1,17 +1,39 @@
-FROM node:12.18-alpine
-
+# Build image
+FROM node:12.18-alpine AS build
 ARG DATABASE_TYPE
-
 ENV DATABASE_URL "postgresql://umami:umami@db:5432/umami" \
     DATABASE_TYPE=$DATABASE_TYPE
+WORKDIR /build
 
-WORKDIR /app
-EXPOSE 3000
+COPY package.json yarn.lock /build/
 
-COPY package.json yarn.lock /app/
-RUN yarn install --frozen-lockfile
+# Install only the production dependencies
+RUN yarn install --production
 
-COPY . /app
+# Cache these modules for production
+RUN cp -R node_modules/ prod_node_modules/
+
+# Install development dependencies
+RUN yarn install
+
+COPY . /build
 RUN yarn build
 
+# Production image
+FROM node:12.18-alpine AS production
+WORKDIR /app
+
+# Copy cached dependencies
+COPY --from=build /build/prod_node_modules ./node_modules
+
+# Copy generated Prisma client
+COPY --from=build /build/node_modules/\.prisma/ ./node_modules/\.prisma/
```

Comment:
not a review, but more a question.

what is the `\` here for? should be `.prisma/`, right?

Reply:
@tim-hub I found that it didn't copy over `.prisma`, I had to escape the `.`. This might be to do with the file path matching rules in Golang 🤔

Reply:
For what it's worth, I've had the same issue with the `@` symbol in path names

Reply:
I've just tested it without escaping, and it seems to work now! I'll remove it in #206